lib/IRGen/IRGenSIL.cpp

```diff
@@ -1991,43 +2014,63 @@ void IRGenSILFunction::visitGraphOperationInst(GraphOperationInst *i) {
   auto *getContextFn = IGM.getTFC_GetGlobalEagerContextFn();
   auto eagerContext = Builder.CreateCall(getContextFn, {});
 
-  // Create a TFE op as in:
-  //   auto* op = TFE_NewOp(ctx, "OpName", status);
+  LLVM_DEBUG(llvm::dbgs() << "Calling TFE_NewOp(ctx, op, status) with op name "
+                          << opInfo.getOperationName() << ".\n");
   auto *TFENewOpFn = IGM.getTFE_NewOpFn();
   auto opNameValAddr = createStringValAddr(IGM, opInfo.getOperationName());
   auto op =
       Builder.CreateCall(TFENewOpFn, {eagerContext, opNameValAddr, status});
   checkOk(status);
 
+  auto addTensorhandleAsOpInput = [=](SILValue tensorHandle) {
+    LLVM_DEBUG(llvm::dbgs()
+               << " Adding input of type " << tensorHandle->getType() << ".\n");
+    auto valueKind = classifyTensorFlowValue(tensorHandle->getType());
+    assert(valueKind != TFValueKind::Nope &&
```

Comment:
You can call `tf::isTensorFlowValue(tensorHandle->getType())`. That function does the same thing as these two lines.

Reply:
Nice. done.